src/components/basic_table/basic_table.tsx

```diff
@@ -606,7 +606,7 @@ export class EuiBasicTable<T = any> extends Component<
 
     if (selection) {
       headers.push(
-        <EuiTableHeaderCellCheckbox key="_selection_column_h" width="24px">
+        <EuiTableHeaderCellCheckbox key="_selection_column_h" width="32px">
```

Comment:
```suggestion
        <EuiTableHeaderCellCheckbox key="_selection_column_h">
```

Looks like you can just remove this width statement. It looks like it gets 32px from `.euiTableHeaderCellCheckbox`

You'll want to update the snaps if you take this change.